tuxemon/event/actions/set_code.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import final
+
+from tuxemon.event.eventaction import EventAction
+from tuxemon.locale import T
+from tuxemon.menu.input import InputMenu
+from tuxemon.tools import open_dialog
+
+
+@final
+@dataclass
+class SetCodeAction(EventAction):
+    """
+    Set a code and checks if it's correct or not.
+    Case Sensible:
+    ATTENTION and AtTenTION are two different words.
+
+    Script usage:
+        .. code-block::
+
+            set_code <code>,<characters>,<variable>
+
+    Script parameters:
+        code: The actual code the player needs to know.
+        characters: The number of characters/digits.
+        variable: Where the result (on/off) is saved.
+
+    """
+
+    name = "set_code"
+    code: str
+    characters: int
+    variable: str
+
+    def check_setcode(self, name: str) -> None:
+        player = self.session.player
+        if self.code == name:
+            player.game_variables[self.variable] = "on"
+            open_dialog(
+                self.session,
+                [T.format("right_code", {"code": name})],
+            )
```

Comment:
I don't think we should have the dialogue hard-coded. Just let the modder trigger the dialogue manually using the output variable.

Reply:
It makes sense, so in the following steps the modder can check the variable by existing actions/conditions.